src/test/java/BaseTest.java

```diff
@@ -1,16 +1,43 @@
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.ie.InternetExplorerDriver;
+import org.testng.annotations.*;
 
 import java.util.concurrent.TimeUnit;
 
+
 public class BaseTest {
-    public WebDriver driver;
+    protected WebDriver driver;
 
-    public void openMainPage() {
-        System.setProperty("webdriver.chrome.driver","src/test/resources/chromedriver_win32/chromedriver.exe");
-        driver = new ChromeDriver();
-        driver.manage().timeouts().implicitlyWait(50, TimeUnit.SECONDS);
-        driver.get("https://price.ua/");
+    @Parameters("browser")
+    @BeforeClass(alwaysRun = true)
+    public void setUp(@Optional("chrome") String browser) throws Exception {
+        initDrivers(browser);
+        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
         driver.manage().window().maximize();
     }
+
+    private void initDrivers(String browser) {
+        switch (browser){
+            case "chrome":
+                System.setProperty("webdriver.chrome.driver", "src/main/resources/properties/chromedriver.exe");
+                driver = new ChromeDriver();
+                break;
+            case "firefox":
+                System.setProperty("webdriver.gecko.driver", "src/main/resources/properties/gecodriver.exe");
+                driver = new FirefoxDriver();
+                break;
+            case "ie":
+                System.setProperty("webdriver.ie.driver", "src/main/resources/properties/IEDriverServer.exe");
+                driver = new InternetExplorerDriver();
+                break;
+            default: throw new IllegalArgumentException("Unknown browser" + browser);
```

Comment:
`String.format()` for strings concatination

Reply:
Done